### DISCUSSION TITLE
Cable Trace Result in Custom Export

### DISCUSSION THREAD

**Discussion by kickofline** [netbox-community/netbox#15023] [Help Wanted!] [answered]
Is there any way that I can show the result of a cable trace (or just what a port is connected to) in a custom export? It shows exactly the data I want in the Interfaces view. 
![image](https://github.com/netbox-community/netbox/assets/31192414/3245007f-620b-4e45-b75c-20fde045c18c)
The cable would run though some patch panels on the way.
My trace:
![image](https://github.com/netbox-community/netbox/assets/31192414/904dbd52-fa19-4b78-9e3f-96b287126f0d)
Currently, I am looping through the link peers to reach the patch panel, but It feels like there is a better way. I can't even reach the final device.
```
output = "Device Name,Peer Device,Peer Interface, Peer2, Interface 2\n"
for device in queryset:
    if device.interfaces.all():
        for interface in device.interfaces.all():
            if interface.link_peers:
                for peer in interface.link_peers:
                    if peer.device: 
                        if peer.device.rear_port.device:
                             if interface2.link_peers:
                                    for peer2 in interface2.link_peers:
                                        if peer2.device:
                                            output += f"\"{device.name}\",\"{peer.device}\",\"{peer.name}\",\"{peer2.device.name}\",\"peer2.name}\"\n"
                        else:	
                            output += f"\"{device.name}\",\"{peer.device}\",\"{peer.name}\"\n"


```
![image](https://github.com/netbox-community/netbox/assets/31192414/1d49eadb-343f-4e01-b070-9667db21dc30)

Thanks for your help!

**Comment by candlerb** [comment, marked as answer]
I am guessing that is the "Connection" column you've highlighted? (Not "Link Peers"?)

[nbshell](https://docs.netbox.dev/en/stable/administration/netbox-shell/) is a good way to explore the available methods and properties interactively.

```
# /opt/netbox/venv/bin/python /opt/netbox/netbox/manage.py nbshell
...
>>> d = Device.objects.get(name="gw2")
>>> i = d.interfaces.get(name="ether6")
>>> help(Interface)
...
>>> i.connected_endpoints
[<Interface: en5>]
>>> i.connected_endpoints[0].device
<Device: MacBook-Pro-4>
```

Note that this is a list, because in general a cable is allowed to have multiple terminations at each end.

**Reply by kickofline** [reply]
Thank you!
